app/components/browse/controls.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Browse
+  class Controls < ViewComponent::Base
+    attr_reader :navigator, :paginator
+
+    # @param navigator [ViewComonent::Base]
+    # @param paginator [ViewComonent::Base]
+    def initialize(navigator:, paginator:)
+      unless navigator.is_a?(ViewComponent::Base) && paginator.is_a?(ViewComponent::Base)
+        raise ArgumentError, 'navigator and paginator must be view components'
+      end
```

Comment:
yeah. it's a trade-off. Incidentally, this is probably an anti-pattern, since we don't really _have_ to restrict it to only `ViewComponent::Base` objects. Because Ruby isn't strongly typed, you could pass in any object you want, and as long as it responds to accordingly, it should be fine.

The only reason I put this in here is so we tell our future selves what this is supposed to do. But, I'd be open to taking it out or coming up with a different way.

Reply:
I think this is totally fine. Like you said, it's good documentation for our future selves.